src/lib/config.js

```diff
@@ -11,13 +13,65 @@ function isRunningTests () {
      process.argv[1].endsWith('mocha'))
 }
 
+function useTestConfig () {
+  return !Config.castBool(process.env.UNIT_TEST_OVERRIDE) && isRunningTests()
+}
+
 function generateDefaultPairs (ledgers) {
   return Utils.getPairs(ledgers).map((pair) => {
     return pair.map(desc => desc.currency + '@' + desc.ledger)
   })
 }
 
+function parseCredentialsEnv () {
+  return JSON.parse(Config.getEnv(envPrefix, 'CREDENTIALS') || '{}')
+}
+
+function parseCredentials () {
+  const credentialsEnv = parseCredentialsEnv()
+
+  return _.reduce(credentialsEnv, (parsed, credentials, ledger) => {
+    const isClientCertCredential = credentials.key !== undefined
+
+    if (isClientCertCredential) {
+      parsed[ledger] = _.assign(credentials, {
+        key: fs.readFileSync(credentials.key),
+        cert: fs.readFileSync(credentials.cert),
+        ca: credentials.cert && fs.readFileSync(credentials.ca)
+      })
+    } else {
+      parsed[ledger] = credentials
+    }
+
+    return parsed
+  }, {})
+}
+
+function validateLocalEnvConfig () {
+  const credentials = parseCredentialsEnv()
+
+  _.forEach(credentials, (credential, ledger) => {
+    if (credential.username === undefined ^ credential.password === undefined) {
```

Comment:
This does work, but `^` is the bitwise XOR operator, not a boolean XOR operator, which means this returns `0` or `1`, which would be a type error if we apply type checking